druid/src/widget/clip_box.rs

```diff
@@ -188,6 +232,16 @@ impl<T, W: Widget<T>> ClipBox<T, W> {
         }
     }
 
+    /// Adjust the viewport to display as much of the target region as is possible.
+    ///
+    /// This will move the viewport the smallest distance that fully shows
+    /// the target region. If the target region is larger than the viewport,
+    /// we will display the portion that fits, prioritizing the portion closest
+    /// to the origin.
+    pub fn pan_to_visible(&mut self, region: Rect) -> bool {
+        self.port.pan_to_visible(region)
```

Comment:
I think you'll need to do `self.child.set_viewport_offset` here, like in `pan_to`

Reply:
oh yea definitely missed that, thanks :)